src/main/java/com/nekomaster1000/infernalexp/world/gen/features/PlantedQuartzFeature.java

```diff
@@ -0,0 +1,98 @@
+package com.nekomaster1000.infernalexp.world.gen.features;
+
+import com.mojang.serialization.Codec;
+import com.nekomaster1000.infernalexp.blocks.PlantedQuartzBlock;
+import com.nekomaster1000.infernalexp.init.IEBlocks;
+import com.nekomaster1000.infernalexp.world.gen.features.config.PlantedQuartzFeatureConfig;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.state.properties.AttachFace;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.ISeedReader;
+import net.minecraft.world.gen.ChunkGenerator;
+import net.minecraft.world.gen.feature.Feature;
+
+import java.util.Random;
+
+public class PlantedQuartzFeature extends Feature<PlantedQuartzFeatureConfig> {
+
+    public PlantedQuartzFeature(Codec<PlantedQuartzFeatureConfig> codec) {
+        super(codec);
+    }
+
+    @Override
+    public boolean generate(ISeedReader worldIn, ChunkGenerator generator, Random random, BlockPos pos, PlantedQuartzFeatureConfig config) {
+        int i = 0;
+        int amount = 15;
+        AttachFace face;
+        Direction direction = Direction.NORTH;
+
+        int pickFacing = random.nextInt(4);
+
+        if (pickFacing <= 1) {
+            face = AttachFace.FLOOR;
+        } else if (pickFacing == 2) {
+            face = AttachFace.WALL;
+            int pickDirection = random.nextInt(4);
+
+            if (pickDirection == 0) {
+                direction = Direction.NORTH;
+            } else if (pickDirection == 1) {
+                direction = Direction.SOUTH;
+            } else if (pickDirection == 2) {
+                direction = Direction.EAST;
+            } else {
+                direction = Direction.WEST;
+            }
+        } else {
+            face = AttachFace.CEILING;
+        }
```

Comment:
I think it would be better if you checked for the placeable faces and then chose from the available faces instead of choosing a random face and hoping that it would generate on a vaild face.

Reply:
How would you propose is the best way to make that check?

Reply:
*responded in discord*